### PR TITLE
chore: attempt to resolve typescript build issues

### PR DESCRIPTION
**Pull request by braxtonchristensen**
When running builds or unit tests in our application, it would fail because the types were not isolated from the JS or functional code.  [see example](https://snack.expo.dev/wzXrlajFx)

Setting type explicitly tells other tools to ignore those imports without the need to compile or add more plugins.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -6,3 +6,6 @@ babel.config.js
 example/react-native.config.js
 example/metro.config.js
 example/babel.config.js
+
+# generated by bob
+lib/
```
Comment (braxtonchristensen) [inline .eslintignore]: Probably don't want to lint build files.

File: package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
   ],
   "scripts": {
     "test": "jest",
-    "typescript": "tsc --noEmit",
+    "typescript": "tsc",
```
Comment (braxtonchristensen) [inline package.json:26]: Let emit happen so you can provide the files listed in the package.json file.

File: src/index.tsx [hunk 3]
```diff
@@ -6,4 +6,5 @@ import type { Operation } from './impl/Operation';
 import type { Zone } from './Zone';
 import type { Campaign } from './Campaign';
 
-export { Condition, Action, AppEvent, Operation, Zone, Campaign, ExtoleImpl as Extole };
+export type { Condition, Action, AppEvent, Operation, Zone, Campaign };
+export { ExtoleImpl as Extole };
```
Comment (braxtonchristensen) [inline src/index.tsx:10]: Separate type exports from implementation exports so other tools know to ignore them.

File: .eslintignore (hunk 1 above)
Comment (daniftodi) [reply .eslintignore]: we have the `lib` directory ignored, it is on line 4

File: package.json (hunk 2 above)
Comment (daniftodi) [inline package.json]: can you please explain the problem with `--noEmit`?

File: package.json [hunk 4]
```diff
@@ -2,10 +2,10 @@
   "name": "@extole/react-native-sdk",
   "version": "0.0.2",
   "description": "Extole React Native SDK",
-  "main": "lib/commonjs/index",
-  "module": "lib/module/index",
+  "main": "lib/commonjs/index.js",
+  "module": "lib/module/index.js",
   "types": "lib/typescript/index.d.ts",
-  "react-native": "src/index",
+  "react-native": "src/index.tsx",
```
Comment (daniftodi) [inline package.json]: can you please explain the current problem?

Comment (braxtonchristensen) [reply package.json]: You are absolutely right. These changes are unnecessary. 
The file extensions were included with the default output of `builder bob`; 
image
I noticed it was outdated, so I updated the dependency and ran to keep the project fresh.

File: package.json (hunk 2 above)
Comment (braxtonchristensen) [reply package.json]: I think this was an artifact of my trying to figure out the build process

File: src/index.tsx (hunk 3 above)
Comment (braxtonchristensen) [reply src/index.tsx:10]: https://github.com/evanw/esbuild/issues/314#issuecomment-668401819

Comment (braxtonchristensen) [reply src/index.tsx:10]: I dont think builder bob is using esbuild but the maintainer calls out "Babel's TypeScript compiler" and I am pretty sure that is what builder bob uses under the hood.

**Comment by daniftodi** [comment]
Thank you @braxtonchristensen for your PR, we are going to release a new version soon that will include the recommendations that you made. 

We don't have yet a mechanism to merge this PR into our code repositories because of some internal procedure that we are going to change in near future.

**Pull request closed without merging**